### PR TITLE
CompositeDistribution: Fix getSingularities and add computeProbability

### PR DESCRIPTION
**Pull request by regislebrun**
Addresses #3308.

This PR fixes two bugs in CompositeDistribution and adds a missing method:

### Bug fixes
- **`getSingularities()`**: previously returned images of *all* interior partition bounds. For the explicit constructor (4-arg), this incorrectly reported singularity images at refinement points inside monotonic regions. Now only returns images of points where the monotonicity actually changes.
- **`computeProbability()`**: constant segments (where g(a)==g(b)) map their antecedent mass to a single value, creating an atom. These atoms were previously dropped for singleton intervals and degenerate cases. Now properly counted.

### New functionality
- **`computeProbability()`**: now computes interval probabilities correctly using a segment-wise summation approach, avoiding Gauss-Kronrod divergence at PDF singularities.

### Minor
- Tests: use `2*math.pi`, exact algebraic values, `TypeError` for constructor argument checks
- Removed redundant `SetSeed(0)` (default is already 0)
- Dropped ChangeLog bug-fix entry (updated at release time)

## Test results
- `cppcheck_CompositeDistribution_std`: PASS
- `pyinstallcheck_CompositeDistribution_std`: PASS
- flake8/doc8: PASS

## Summary by CodeRabbit

* **New Features**
  * Added interval probability calculations for composite distributions, including accurate handling of discrete atoms at segment boundaries and degenerate continuous intervals.
  * Enhanced singularity reporting to include stationary critical points and monotonicity changes.
  * Added documentation for automatic partitioning, solver configuration, probability calculations, and singularity behavior.

* **Bug Fixes**
  * Improved probability accuracy across monotonic distribution segments and boundary cases.

* **Tests**
  * Expanded coverage for distributions, singularities, probabilities, serialization, configuration, ranges, and edge cases.

### CODE REVIEW

**Comment by regislebrun** [comment]
@jschueller I add pushed the fixes directly in master by mistake. I took all your remarks into account (I hope so) before to close the previous PR and open this one. Sorry for the burden.

**Review by jschueller — APPROVED** [review]

File: python/src/CompositeDistribution_doc.i [hunk 1]
```diff
@@ -43,6 +43,23 @@ Notes
...
+
+The roots of :math:`g'` are located by the solver on a regular grid of
+:math:`CompositeDistribution-StepNumber` subintervals of the support of
+:math:`distX` (defaults to 256). The solver and its tolerances can be changed
+with :meth:`setSolver`, using as default absolute tolerance
+:math:`CompositeDistribution-SolverEpsilon` rescaled by the antecedent range
+width.
+
+The range of :math:`distY` is the interval between the minimum and the maximum
+of the values :math:`v[k]`.
```
Comment (jschueller) [inline python/src/CompositeDistribution_doc.i:61]: this missing doc for singularities is enough to close #3282, could you mention it it the commit message for automatic closing ?

**Comment by regislebrun** [comment]
done

File: python/src/CompositeDistribution_doc.i (hunk 1 above)
Comment (regislebrun) [reply python/src/CompositeDistribution_doc.i:61]: done

File: python/src/CompositeDistribution_doc.i [hunk 2]
```diff
@@ -43,6 +43,23 @@ Notes
...
 Its first moments are obtained by numerical integration.
 
+The two-argument constructor determines the monotonicity partition
+:math:`(a_0,\ldots,a_N)` automatically, by solving
+:math:`g'(x)=0` on the support of :math:`distX`. The four-argument
+constructor lets the user specify this partition, when the user knows a priori
+the monotonicity intervals of :math:`g`, or when :math:`g` is not smooth enough
+for its derivatives to be computed numerically by the automatic procedure.
+
+The roots of :math:`g'` are located by the solver on a regular grid of
```
Comment (adutfoy) [inline python/src/CompositeDistribution_doc.i:53]: These keys should be detailed as done for the other classes!

File: python/src/CompositeDistribution_doc.i [hunk 3]
```diff
@@ -43,6 +43,23 @@ Notes
 -----
 Its first moments are obtained by numerical integration.
 
+The two-argument constructor determines the monotonicity partition
+:math:`(a_0,\ldots,a_N)` automatically, by solving
+:math:`g'(x)=0` on the support of :math:`distX`. The four-argument
```
Comment (adutfoy) [inline python/src/CompositeDistribution_doc.i:48]: Instead of :math:`g`, use :math:`\model`

File: python/src/CompositeDistribution_doc.i [hunk 4]
```diff
@@ -130,3 +147,61 @@ Returns
...
+
+Returns
+-------
+p : float
+    The probability of *interval*.
+
+Notes
+-----
+This method is implemented as a segment-wise summation over the monotonicity
+intervals of :math:`g`, using the precomputed CDF values of :math:`distX` at
```
Comment (adutfoy) [inline python/src/CompositeDistribution_doc.i:169]: use :math:`\model` instead of g!

**Review by adutfoy — CHANGES_REQUESTED** [review]

File: python/src/CompositeDistribution_doc.i (hunk 2 above)
Comment (regislebrun) [reply python/src/CompositeDistribution_doc.i]: Fixed: ResourceMap keys are now detailed as a canonical bullet list (`CompositeDistribution-StepNumber`, default `256`; `CompositeDistribution-SolverEpsilon`, default `1.0e-14`, with types), merged into the existing Notes section. See commit e7cfa1a57.

File: python/src/CompositeDistribution_doc.i (hunk 3 above)
Comment (regislebrun) [reply python/src/CompositeDistribution_doc.i]: Fixed: the constructor paragraph now uses the :math:`\model` notation macro instead of :math:`g`. See commit e7cfa1a57.

File: python/src/CompositeDistribution_doc.i (hunk 4 above)
Comment (regislebrun) [reply python/src/CompositeDistribution_doc.i]: Fixed: computeProbability Notes now uses :math:`\model` instead of g. See commit e7cfa1a57.

**Review by adutfoy — APPROVED** [review]

**Comment by regislebrun** [comment]
@jschueller Two ok, no ko, go to master?

**Comment by jschueller** [comment]
this branch needs to be rebased

File: lib/src/Uncertainty/Distribution/CompositeDistribution.cxx [hunk 5]
```diff
@@ -446,6 +447,208 @@ Scalar CompositeDistribution::computeCDF(const Point & point) const
...
+          const Scalar atomMass = antecedent_.computePDF(b);
+          if (atomMass > 0.0) pHi -= atomMass;
+        }
+      }
+      else
+      {
+        const Point t(1, solver_.solve(function_, hi, a, b, fA, fB));
+        pHi = antecedent_.computeCDF(t);
+      }
+      probability += pLo - pHi;
```
[the remark this answers is not in the document]
Comment (regislebrun) [reply lib/src/Uncertainty/Distribution/CompositeDistribution.cxx]: Fixed: replaced the ad hoc endpoint branches with clamped CDF(t) for upper bounds and CDF(t-) for lower bounds, so interior atoms are counted once and lo==fB no longer includes the full segment. See commit ed8252721.

**Comment by regislebrun** [comment]
@jschueller Rebased onto current upstream/master (base 76c2ad5aa, linear history with no merge commits). Latest push ed8252721 fixes the last review thread (computeProbability endpoint atoms + regression tests). CI is running on that commit; I will confirm here once it is green.

**Comment by regislebrun** [comment]
@jschueller This one looks ready: coderabbit is happy, two approvals, no conflict, no thumb down.
